.github/workflows/sphinx_asdf_ci.yml

```diff
@@ -0,0 +1,150 @@
+          - name: Python 3.8
+            python-version: 3.8
+            os: ubuntu-latest
+            script: |
+              python -m pip install --upgrade pip
+              python -m pip install .[tests]
+              python -m pip install sphinx
+            toxenv: py38
+
+          - name: Python 3.7
+            python-version: 3.7
+            os: ubuntu-latest
+            script: |
+              python -m pip install --upgrade pip
+              python -m pip install .[tests]
+              python -m pip install sphinx
+            toxenv: py37
+
+          - name: Python 3.6
+            python-version: 3.6
+            os: ubuntu-latest
+            script: |
+              python -m pip install --upgrade pip
+              python -m pip install .[tests]
+              python -m pip install sphinx
+            toxenv: py36
+
+          - name: Python 3.9, Windows
+            python-version: 3.9
+            os: windows-latest
+            script: |
+              python -m pip install --upgrade pip
+              python -m pip install .[tests]
+              python -m pip install sphinx
+            toxenv: py39
+
+          - name: Python 3.9, MacOS
+            python-version: 3.9
+            os: macos-latest
+            script: |
+              python -m pip install --upgrade pip
+              python -m pip install .[tests]
+              python -m pip install sphinx
+            toxenv: py39
+
+          - name: ASDF Standard Build
+            python-version: 3.9
+            os: ubuntu-latest
+            script: |
+              python -m pip install --upgrade pip
+              python -m pip install sphinx
+            toxenv: asdf-standard
+
+          - name: Code Style Checks
+            python-version: 3.9
+            os: ubuntu-latest
+            script: |
+              python -m pip install --upgrade pip
+              python -m pip install flake8
+            toxenv: style
+
+          - name: Twine
+            python-version: 3.9
+            os: ubuntu-latest
+            script: |
+              python -m pip install --upgrade pip
+              python -m pip install twine
+            toxenv: twine
+
+          - name: Check Documents
+            python-version: 3.9
+            os: ubuntu-latest
+            script: |
+              python -m pip install --upgrade pip
+              python -m pip install .[tests]
+              python -m pip install collective.checkdocs
+              python -m pip install sphinx
+            toxenv: checkdocs
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install Dependencies
+        run: |
+          ${{ matrix.script }}
+      - name: Install tox
+        run: |
+          python -m pip install --upgrade pip
```

Comment:
Since the steps of this workflow already upgrades `pip`

https://github.com/asdf-format/sphinx-asdf/pull/23/files#diff-ee2ac86122d1cc8a8ce89f4c40efc1c86fbbeb72a3e328e9a7fd2027be3d4c4aR138

we probably don't need to do it in each job in the matrix.